tools/batch_correction/determine_bc.xml

```diff
@@ -0,0 +1,222 @@
+<tool id="Determine_BC" name="Determine_batch_correction" version="3.0.0+galaxy0" profile="23.0">
+  <description>to choose between linear, lowess and loess methods</description>
+  
+  <macros>
+    <import>macros.xml</import>
+  </macros>
+  
+  <expand macro="requirements">
+    <requirement type="package" version="3.0.0">libgfortran</requirement>
+  </expand>
+ 
+  <command detect_errors="aggressive"><![CDATA[
+    Rscript $__tool_directory__/batch_correction_3Lwrapper.R
+      analyse "determine_bc"
+      sampleMetadata "$sampleMetadata"
+      dataMatrix "$dataMatrix"
+      ref_factor "$ref_factor"
+      #if str($advance.option ) == 'show':
+        span $advance.span
+      #else:
+        span "none"
+      #end if
+      out_graph_pdf "$out_graph_pdf"
+      out_preNormSummary "$out_preNormSummary"
+      @SM_CUSTOM@
+  ]]></command>
+  
+  <inputs>
+    <param name="dataMatrix" type="data" label="Data matrix file " help="" format="tabular" />
+    <param name="sampleMetadata" type="data" label="Sample metadata file " help="must contain at least the three following columns: 'batch' + 'injectionOrder' + 'sampleType'" format="tabular" />
+    <expand macro="sm_customisation"/>
+    <expand macro="foi"/>
+    <conditional name="advance">
+        <param name="option" type="select" label="Advanced options" help="should only be put at 'show' if you have good understanding of the span parameter and are willing to adjust it">
```

Comment:
this conditional seems to be not needed, you can make the span param just optional?